dm/syncer/validator_checkpoint.go

```diff
@@ -383,18 +425,18 @@ func (c *validatorPersistHelper) loadPersistedDataRetry(tctx *tcontext.Context)
 		c.L.Error("failed load persisted data after retry", zap.Int("retry-times", i), zap.Error(err))
 		return nil, err
 	}
-	return ret.(*persistedData), err
+	return ret.(*persistedData), nil
 }
 
 func (c *validatorPersistHelper) loadPersistedData(tctx *tcontext.Context) (*persistedData, error) {
 	var err error
 	data := &persistedData{}
-	data.checkpoint, err = c.loadCheckpoint(tctx)
+	data.checkpoint, data.rev, err = c.loadCheckpoint(tctx)
 	if err != nil {
 		return data, err
 	}
 
-	data.pendingChanges, data.rev, err = c.loadPendingChange(tctx)
+	data.pendingChanges, err = c.loadPendingChange(tctx, data.rev)
```

Comment:
Is it possible that checkpoint's rev is different with loadPendingChange's rev?

Reply:
as explained in the pr desc, now we use revision field to link checkpoint and pending data

on `persist`, pending change is saved first, then checkpoint, and they save using same `revision` . on success the revision of checkpoint equals to the revision in pending change(there may still have pending data of other revision, but it's trash data that need to clean up). so on loading, we have to use the save `revision` from `checkpoint` table to load data from `pending change` table.